bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/repository/simple/DownloadJob.java

```diff
@@ -56,8 +58,7 @@ protected IStatus run(IProgressMonitor jobMonitor) {
 			if (masterMonitor.isCanceled())
 				return Status.CANCEL_STATUS;
 			// process the actual request
-			SubMonitor subMonitor = SubMonitor.convert(masterMonitor.slice(1), 1);
-			IStatus status = repository.getArtifact(request, subMonitor);
+			IStatus status = repository.getArtifact(request, new ThreadSafeProgressMonitor(1));
```

Comment:
I though that there are multiple jobs, so doe not need the "master monitor" already be thread safe?!?

Reply:
Yes, this is why I asked the question I asked yesterday.  The monitor here is used only in this thread I believe.